dask_ndfilters/order.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+
+
+import functools
+import inspect
+import numbers
+import re
+
+import numpy
+import scipy.ndimage.filters
+
+
+def _get_footprint(ndim, size=None, footprint=None):
+    # Verify that we only got size or footprint.
+    if size is None and footprint is None:
+        raise RuntimeError("Must provide either size or footprint.")
+    if size is not None and footprint is not None:
+        raise RuntimeError("Provide either size or footprint, but not both.")
+
+    # Get a footprint based on the size.
+    if size is not None:
+        if isinstance(size, numbers.Integral):
+            size = ndim * (size,)
+        footprint = numpy.ones(size, dtype=bool)
+
+    # Validate the footprint.
+    if footprint.ndim != ndim:
+        raise RuntimeError(
+            "The footprint must have the same number of dimensions as"
+            " the array being filtered."
+        )
+    if footprint.size == 0:
+        raise RuntimeError("The footprint must have only non-zero dimensions.")
+
+    # Convert to Boolean.
+    footprint = (footprint != 0)
+
+    return footprint
+
+
+def _get_origin(footprint, origin=0):
+    size = numpy.array(footprint.shape)
+    ndim = footprint.ndim
+
+    if isinstance(origin, numbers.Real):
+        origin = ndim * (origin,)
+
+    origin = numpy.array(origin)
+    origin = numpy.fix(origin).astype(int)
```

Comment:
Probably the easiest thing is to turn this into an error outright. Can always soften it later if we decide to.